### DISCUSSION TITLE
ColorShape and ColorMultiShape

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#10421] [2D]
It would be useful to add simple 2d shapes to godot.
I am macing grid that uses transparent rectangles, but overlaying areas have more opacity. And it would be easer to use more shapes like circle and other simple shapes.

**Comment by Calinou** [comment]
This already exists as the ColorRect and Polygon2D nodes (and Line2D as well if you need that).

**Reply by ALapidis** [reply]
The problem is that the polygon2D does not respect the same layout options as the other 2D nodes. It's position is absolute rather then as an element in a container like ColorRect. I believe ghost was asking for more versital shapes that can be used which behave similar to ColorRect.
